libpromises/expand.c

```diff
@@ -259,6 +259,12 @@ PromiseResult ExpandPromise(EvalContext *ctx, const Promise *pp,
 {
     if (!IsDefinedClass(ctx, pp->classes))
     {
+        Log(LOG_LEVEL_DEBUG,
+            "Skipping %s promise expansion with promiser '%s' due to class guard '%s::' (pass %d)",
+            pp->parent_section->promise_type,
+            pp->promiser,
+            pp->classes,
```

Comment:
Might as well add the `assert(pp != NULL)` to make LGTM happy.

Reply:
Done